src/ol/format/KML.js

```diff
@@ -2713,6 +2703,18 @@ function writePlacemark(node, feature, objectStack) {
   pushSerializeAndPop(context, PLACEMARK_SERIALIZERS,
     OBJECT_PROPERTY_NODE_FACTORY, values, objectStack, orderedKeys);
 
+  properties = feature.getProperties();
```

Comment:
Instead of reading the properties again here, you could keep the order as it was, and only call `pushSerializeAndPop(context, PLACEMARK_SERIALIZERS, EXTENDEDDATA_NODE_FACTORY, [namesAndValues], objectStack)` here instead of above.

Reply:
Thanks for your work on this! It would be great if you could address the above comment.

Reply:
Thanks for the review and the reactivity! :smile: